docs/topics/native/native-app-with-c-and-libcurl.md

```diff
@@ -0,0 +1,260 @@
+
+## Create a definition file
+
+When writing native applications, you often need access to certain functionality that is not included in the [Kotlin
+standard library](https://kotlinlang.org/api/latest/jvm/stdlib/), such as making HTTP requests, reading and writing from
+disk, and so on.
+
+Kotlin/Native helps consume standard C libraries, opening up an entire ecosystem of functionality that exists
+for pretty much anything you may need. Kotlin/Native is already shipped with a set of prebuilt [platform libraries](native-platform-libs.md),
+which provide some additional common functionality to the standard library.
+
+An ideal scenario for interop is to call C functions as if you are calling Kotlin functions, following the same
+signature and conventions. This is when the `cinterop` tool comes in handy. It takes a C library and generates the
+corresponding Kotlin bindings, so that the library can be used as if it were a Kotlin code.
+
+To generate these bindings, let's create a library definition `.def` file that contains some information about
+the necessary headers. In our app, you'll need the `libcurl` library to make some HTTP calls. To create a
+definition file:
+
+1. Select the `src` folder and create a new directory with **File | New | Directory**.
+2. Name new directory **nativeInterop/cinterop**. This is the default convention for header file locations, though it can
+   be overridden in the `build.gradle` file if you use a different location.
+3. Select this new subfolder and create a new `libcurl.def` file with **File | New | File**.
+4. Update your file with the following code:
+
+```c
+headers = curl/curl.h
+headerFilter = curl/*
+
+compilerOpts.linux = -I/usr/include -I/usr/include/x86_64-linux-gnu
+linkerOpts.osx = -L/opt/local/lib -L/usr/local/opt/curl/lib -lcurl
+linkerOpts.linux = -L/usr/lib/x86_64-linux-gnu -lcurl
+```
+
+* The first entry is `headers`, the list of header files to generate Kotlin stubs for. You can add multiple files to
+this entry, separating each with a `\` on a new line. In our case, it's only `curl.h`. The referenced files need to be
+relative to the folder where the definition file is or be available on the system path (in our case, it's `/usr/include/curl`).
+* The second line is the `headerFilter`. It shows what exactly is included. In C, all the headers are also included when
+one file references another one with the `#include` directive. Sometimes it's not necessary, and you can add this
+parameter [using glob patterns](https://en.wikipedia.org/wiki/Glob_(programming)) to fine-tune things.
+
+> `headerFilter` is an optional argument and is mostly used when the library is installed as a system library. You don't
+> want to fetch external dependencies (such as system `stdint.h` header) into the interop library. It may be important to
+> optimize the library size and fix potential conflicts between the system and the provided Kotlin/Native compilation
+> environment.
+>
+{type="note"}
```

Comment:
As for me, two notes (one after another) look visually redundant. Can't we rewrite this make a separate para with this text?
Or any other ideas?